friends/src/App.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import FriendList from './components/FriendList';
+import Form from './components/Form';
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      friends: [],
+    }
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:5000/friends')
+      .then(response => {
+        console.log(response);
+        this.setState({ friends: response.data })
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }
+
+  addFriend = friendData => {
+    axios
+      .post('http://localhost:5000/friends', friendData)
+      .then(response => {
+        this.setState({
+          friends: response.data,
+        });
+      })
+      .catch(error => {
+        console.log(error);
+      })
+  }
+
+  updateFriend = (friendData, id) => {
+    axios
+      .put(`http://localhost:5000/friends/${id}`, friendData)
+      .then(response => {
+        this.setState({
+          friends: response.data,
+        });
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }
+
+  deleteFriend = (id) => {
+    axios
+      .delete(`http://localhost:5000/friends/${id}`)
+      .then(response => {
+        this.setState({
+          friends: response.data
+        })
```

Comment:
You could use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) to make this slightly cleaner:
```javascript
// ...
.then(({ data }) => {
  this.setState({
    friends: data
  })
})
.catch(console.log)
```